mfai/pytorch/models/llms/cross_attention.py

```diff
@@ -62,29 +80,73 @@ def __init__(
             vocab_size=vocab_size,
         )
 
-        # Initialize our resnet50 vision encoder
-        num_channels_viz = (
-            settings.vision_input_shape[2] * settings.vision_input_shape[3]
-        )
-        self.vision_encoder = ResNet50MLM(
-            num_channels=num_channels_viz,
-            num_classes=settings.emb_dim,
-            settings=ResNet50MLMSettings(num_tokens=settings.num_tokens_vision),
-        )
+        self.vision_encoder: WeatherProjector | ResNet50MLM | VitEncoder
+
+        if self.settings.vision_encoder == "linear":
+            input_dims = (
+                settings.vision_input_shape[0],
+                settings.vision_input_shape[1],
+                settings.vision_input_shape[-1],
+            )
+            s = WeatherProjectorSettings(
+                input_dims=input_dims,
+                embedding_dim=self.settings.emb_dim,
+                patch_size=self.settings.patch_size,
+            )
+            self.vision_encoder = WeatherProjector(settings=s)
+
+        elif self.settings.vision_encoder == "resnet50":
+            self.vision_encoder = ResNet50MLM(
+                num_channels=settings.vision_input_shape[3],
```

Comment:
May be use the same syntax as for "linear" and "vit":
```suggestion
                num_channels=settings.vision_input_shape[-1],
```